### PR TITLE
Support nested member-access in filter expressions (#392)

### PR DESCRIPTION
**Pull request by ltwlf**
## Summary
- Function-based identity keys returning nested paths (e.g. `positionNumber.value`) now produce `@.positionNumber.value` (RFC 9535 nested dot-notation) instead of `@['positionNumber.value']` (literal property)
- String-based identity keys with dots (literal property names like `c.d`) still use bracket notation `@['c.d']`
- Adds `embeddedKeyIsPath` flag to `IChange` to distinguish the two cases
- Updates atom layer parser/builder to support nested member-access
- Bumps to `5.0.0-alpha.6`

## Test plan
- [x] 343 tests pass, lint/typecheck/build clean
- [x] Python json-atom-py updated and 743 tests pass
- [x] Cross-lib round-trip verified for nested paths

### CODE REVIEW

File: src/jsonDiff.ts [hunk 1]
```diff
@@ -567,12 +570,15 @@ const compareArray = (oldObj: any, newObj: any, path: any, keyPath: any, options
...
+    const resolvedKey = isFunctionKey ? uniqKey(newObj[0], true) : uniqKey;
     return [
       {
         type: Operation.UPDATE,
         key: getKey(path),
-        embeddedKey: typeof uniqKey === 'function' && uniqKey.length === 2 ? uniqKey(newObj[0], true) : uniqKey,
+        embeddedKey: resolvedKey,
+        ...(isFunctionKey && typeof resolvedKey === 'string' && resolvedKey.includes('.') ? { embeddedKeyIsPath: true } : {}),
         changes: diffs
       }
```
[the remark this answers is not in the document]
Comment (ltwlf) [reply src/jsonDiff.ts]: Updated — nested property resolution is now supported in apply/revert via `resolveProperty` helper (1a50700). Also fixed the empty array edge case with `oldObj[0]` fallback. Apply round-trip test added and passing.

File: src/jsonDiff.ts [hunk 2]
```diff
@@ -656,13 +662,20 @@ const removeKey = (obj: any, key: any, embeddedKey: any) => {
...
+
 const indexOfItemInArray = (arr: any[], key: any, value: any) => {
   if (key === '$value') {
     return arr.indexOf(value);
   }
   for (let i = 0; i < arr.length; i++) {
     const item = arr[i];
-    if (item && item[key] ? item[key].toString() === value.toString() : undefined) {
+    const resolved = resolveProperty(item, key);
+    if (item && resolved ? resolved.toString() === value.toString() : undefined) {
```
[the remark this answers is not in the document]
Comment (ltwlf) [reply src/jsonDiff.ts]: Fixed in bf00e94 — uses `resolved != null` and `String()` comparison throughout.

File: src/jsonDiff.ts [hunk 3]
```diff
@@ -152,21 +154,22 @@ const revertChangeset = (obj: any, changeset: Changeset) => {
...
       path = updatedPath;
       if (atomicChange) {
         return atomicChange;
       }
     } else {
       path = append(path, obj.key);
     }
-    return atomizeChangeset(obj.changes || obj, path, obj.embeddedKey);
+    return atomizeChangeset(obj.changes || obj, path, obj.embeddedKey, obj.embeddedKeyIsPath);
   } else {
```
[the remark this answers is not in the document]
Comment (ltwlf) [reply src/jsonDiff.ts]: Fixed in bf00e94 — `unatomizeChangeset` sets `embeddedKeyIsPath` when the regex matches dot-notation filter key (result[2] with dots).

File: src/jsonDiff.ts [hunk 4]
```diff
@@ -273,9 +276,12 @@ const unatomizeChangeset = (changes: IAtomicChange | IAtomicChange[]) => {
           let key: string;
           let embeddedKey: string;
           let arrKey: string | number;
+          let isPath: boolean | undefined;
           if (result[1]) {
             key = result[1];
             embeddedKey = result[3] || result[2] || '$value';
+            // Dot-notation key (result[2]) with dots = nested path
+            isPath = !result[3] && result[2]?.includes('.') || undefined;
             arrKey = result[4];
```
[the remark this answers is not in the document]
Comment (ltwlf) [reply src/jsonDiff.ts]: Fixed in 344b0e0 — regex now handles doubled-quote escaping in bracket keys and filter values. Round-trip tests added for both cases.

File: src/jsonDiff.ts [hunk 5]
```diff
@@ -262,21 +265,22 @@ const unatomizeChangeset = (changes: IAtomicChange | IAtomicChange[]) => {
...
           let arrKey: string | number;
+          let isPath: boolean | undefined;
           if (result[1]) {
             key = result[1];
-            embeddedKey = result[3] || result[2] || '$value';
-            arrKey = result[4];
+            // Unescape doubled quotes in bracket keys and filter values
+            embeddedKey = (result[3]?.replace(/''/g, "'") || result[2] || '$value');
+            // Dot-notation key (result[2]) with dots = nested path
+            isPath = !result[3] && result[2]?.includes('.') || undefined;
```
[the remark this answers is not in the document]
Comment (ltwlf) [reply src/jsonDiff.ts]: Fixed in 94e8138 — uses `NESTED_PATH_RE.test(result[2])` instead of `includes('.')`.

File: src/jsonDiff.ts [hunk 6]
```diff
@@ -656,13 +665,22 @@ const removeKey = (obj: any, key: any, embeddedKey: any) => {
   }
 };
 
-const indexOfItemInArray = (arr: any[], key: any, value: any) => {
+/** Resolve a property on an object. When isPath is true, traverses nested dot-separated segments. */
+const resolveProperty = (obj: any, key: string, isPath?: boolean): any => {
+  if (obj == null) return undefined;
```
[the remark this answers is not in the document]
Comment (ltwlf) [reply src/jsonDiff.ts]: Fixed in a95ddf3 — `resolveProperty` now accepts `any` key type and falls back to direct property access for non-strings.
